src/impl/autoplay.c

```diff
@@ -320,6 +327,153 @@ void game_runner_start(const AutoplayWorker *autoplay_worker,
   }
 }
 
+// Monte Carlo simulation-based move selection for autoplay
+// This function is defined here instead of gameplay.c to avoid circular
+// dependency
+static Move *get_top_computer_move(Game *game, int movegen_thread_index,
+                                   int sim_threads, MoveList *move_list,
+                                   int sim_plies, int sim_num_plays,
+                                   int sim_max_iterations,
+                                   int sim_min_play_iterations,
+                                   double sim_stop_cond_pct, uint64_t sim_seed,
+                                   WinPct *win_pcts) {
+  // If sim_plies is 0 or bag is empty, fall back to equity-based move
+  if (sim_plies <= 0 || bag_is_empty(game_get_bag(game))) {
+    return get_top_equity_move(game, movegen_thread_index, move_list);
+  }
+
+  // Generate moves for simulation
+  const MoveGenArgs args = {
+      .game = game,
+      .move_list = move_list,
+      .move_record_type = MOVE_RECORD_ALL,
+      .move_sort_type = MOVE_SORT_EQUITY,
+      .override_kwg = NULL,
+      .thread_index = movegen_thread_index,
+      .eq_margin_movegen = 0,
+  };
+  generate_moves(&args);
+
+  // Sort the move list (convert from heap to sorted descending array)
+  move_list_sort_moves(move_list);
+
+  // Remove pass move if it exists (it's always first if present due to 0
+  // equity)
+  int num_moves = move_list_get_count(move_list);
+  if (num_moves > 0 &&
+      move_get_type(move_list_get_move(move_list, 0)) == GAME_EVENT_PASS) {
+    // Shift all moves down by one to remove the pass
+    for (int i = 0; i < num_moves - 1; i++) {
+      move_copy(move_list_get_move(move_list, i),
+                move_list_get_move(move_list, i + 1));
+    }
+    move_list->count--;
+    num_moves--;
+  }
+
+  // Limit the number of plays to sim
+  if (num_moves > sim_num_plays) {
+    // Keep only the top sim_num_plays moves by truncating the count
+    // (moves are already sorted by equity from generate_moves)
+    move_list->count = sim_num_plays;
+  }
+
+  // Create sim results for this simulation
+  SimResults *sim_results = sim_results_create();
```

Comment:
We should try to create these at the beginning on a per-thread basis instead of every time during movegen.